src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,366 @@
+    { intros a ha,
+      apply ne_of_lt hμ.measure_univ_lt_top,
+      rw [eq_top_iff, ← ha, outer_measure.measure_of_eq_coe, coe_to_outer_measure],
+      exact measure_mono (set.subset_univ _) }
+  end }
+
+lemma of_measure_apply_measurable {μ : measure α} [finite_measure μ]
+  {i : set α} (hi : measurable_set i) :
+  of_measure μ i = (μ i).to_real :=
+if_pos hi
+
+lemma of_measure_apply_not_measurable {μ : measure α} [finite_measure μ]
+  {i : set α} (hi : ¬ measurable_set i) :
+  of_measure μ i = 0 :=
+if_neg hi
+
+variables {M : Type*} [add_comm_group M] [topological_space M]
+
+/-- The zero signed measure. -/
+def zero : vector_measure α M :=
+⟨0, rfl, λ _ _, rfl, λ _ _ _, has_sum_zero⟩
+
+instance : has_zero (vector_measure α M) := ⟨zero⟩
+instance : inhabited (vector_measure α M) := ⟨0⟩
+
+@[simp]
+lemma zero_apply (i : set α) : (0 : vector_measure α M) i = 0 := rfl
+
+section
+
+variables [topological_add_group M]
+
+/-- The negative of a vector measure is a vector measure. -/
+def neg (v : vector_measure α M) : vector_measure α M :=
+{ measure_of := -v,
+  empty := by simp,
+  not_measurable := λ _ hi, by simp [v.of_not_measurable_set hi],
+  m_Union := λ f hf₁ hf₂,
+    has_sum.neg $ v.has_sum_of_disjoint_Union hf₁ hf₂ }
+
+/-- The sum of two vector measure is a vector measure. -/
+def add (v w : vector_measure α M) : vector_measure α M :=
+{ measure_of := v + w,
+  empty := by simp,
+  not_measurable := λ _ hi,
+    by simp [v.of_not_measurable_set hi, w.of_not_measurable_set hi],
+  m_Union := λ f hf₁ hf₂,
+    has_sum.add (v.has_sum_of_disjoint_Union hf₁ hf₂)
+      (w.has_sum_of_disjoint_Union hf₁ hf₂) }
+
+instance : has_add (vector_measure α M) := ⟨add⟩
+instance : has_neg (vector_measure α M) := ⟨neg⟩
+
+@[simp]
+lemma neg_apply {v : vector_measure α M} (i : set α) :
+  (-v) i = - v i := rfl
+
+@[simp]
+lemma add_apply {v w : vector_measure α M} (i : set α) :
+  (v + w) i = v i + w i := rfl
+
+instance : add_comm_group (vector_measure α M) :=
+{ add := (+), zero := (0),
+  neg := vector_measure.neg,
+  add_assoc := by { intros, ext i; simp [add_assoc] },
+  zero_add := by { intros, ext i; simp },
+  add_zero := by { intros, ext i; simp },
+  add_comm := by { intros, ext i; simp [add_comm] },
+  add_left_neg := by { intros, ext i, simp } } .
+
+end
+
+/-- Given two finite measures `μ, ν`, `of_sub_measure μ ν` is the signed measure
+corresponding to the function `μ - ν`. -/
+def of_sub_measure (μ ν : measure α) [hμ : finite_measure μ] [hν : finite_measure ν] :
+  signed_measure α :=
+of_measure μ + - of_measure ν
+
+lemma of_sub_measure_apply {μ ν : measure α} [finite_measure μ] [finite_measure ν]
+  {i : set α} (hi : measurable_set i) :
+  of_sub_measure μ ν i = (μ i).to_real - (ν i).to_real :=
+begin
+  rw [of_sub_measure, add_apply, neg_apply, of_measure_apply_measurable hi,
+      of_measure_apply_measurable hi, sub_eq_add_neg]
+end
+
+section
+
+variables {R : Type*} [ring R] [module R M]
```

Comment:
Does the `has_scalar` instance work for weaker typeclasses than `module R M`?

Reply:
sadly `smul` uses `has_sum.smul` which only works for modules

Reply:
I get the feeling `has_sum.smul` would work for `distrib_mul_action`s too, but maybe that's out of scope.

Reply:
It does seem to work with `distrib_mul_action`. Would you like to make a PR for this?

Reply:
#8312